src/snowflake/snowpark/mock/_plan.py

```diff
@@ -319,6 +320,14 @@ def handle_function_expression(
     else:
         func_name = exp.name.lower()
 
+    udf_name = exp.name.split(".")[-1]
+    # If udf name in the registry then this is a udf, not an actual function
+    if udf_name in analyzer.session.udf._registry:
+        exp.udf_name = udf_name
+        return handle_udf_expression(
+            exp, input_data, analyzer, expr_to_alias, current_row
+        )
+
```

Comment:
nit: just curious, can a udf be defined using the same signature of a built-in function, what would happen?

Reply:
I tried this out on live and it looks like if you shadow a builtin it it will favor the builtin over your udf. I can change this so that the priority matches.